src/phoenix/server/api/subscriptions.py

```diff
@@ -484,29 +491,35 @@ async def chat_completion(
                 elif isinstance(chunk, ToolCallChunk):
                     yield chunk
                     tool_call_chunks[chunk.id].append(chunk)
-
-            span.set_status(StatusCode.OK)
-
-            span.set_attributes(
-                dict(
-                    chain(
-                        _output_value_and_mime_type(response_chunks),
-                        _llm_output_messages(text_chunks, tool_call_chunks),
-                    )
+                else:
+                    assert_never(chunk)
+            status_code = StatusCode.OK
+        except Exception as error:
+            end_time = cast(datetime, normalize_datetime(dt=local_now(), tz=timezone.utc))
+            status_code = StatusCode.ERROR
+            status_message = str(error)
+            events.append(
+                SpanException(
+                    timestamp=end_time,
+                    message=status_message,
+                    exception_type=type(error).__name__,
+                    exception_escaped=False,
+                    exception_stacktrace=format_exc(),
+                )
+            )
+            yield ChatCompletionSubscriptionError(message=status_message)
```

Comment:
don't know if this matters as much on non SaaS, we could put a prefix on the error if it came from the llm provider, just to inform the user that the error did not happen in our system but happened in a third party api e.g., OpenAI API Error: status_message. Have to parse out where the error comes from which can be a pain though

Reply:
I think this is a good idea. We don't currently have a display name in the backend for the providers, just a provider key (e.g., `openai` without the nice casing). Let's punt until we have.